hw/top_earlgrey/data/top_earlgrey.sv.tpl

```diff
@@ -240,6 +240,8 @@ module top_${top["name"]} #(
     .irq_timer_i          (intr_rv_timer_timer_expired_0_0),
     .irq_external_i       (irq_plic),
     // escalation input from alert handler (NMI)
+    // TODO: need to put the escalation receiver
```

Comment:
is there any reason this has to be the same domain? 
might make sense to define the receiving scheme also.

Reply:
yes, only alert senders support asynchronous clock crossings at the moment.
escalation sender/receiver pairs must be synchronous.

Reply:
i.e., this uses the `prim_esc_sender` / `prim_esc_receiver` pair and the protocol currently does not support asynchronous transitions...

Reply:
should we consider adding async handling directly in there? As an example, all the modules that handle reset, even life cycle controller for example, may not be in the same domain as the processor, which would be the recipient of the NMI.

Technically we can put the domain crossing after the receiver, but that would imply that always on modules may take on the fast clock.  Even if the logic cone is small..it could still be an always on clock tree on faster clocks.

Reply:
hm... yeah that would make sense. we can either add the synchronization before or after.

before would mean a change to the escalation signalling protocol which means there is DV impact, but integration would be cleaner. putting the sync after would probably be simpler to add, but as you mentioned we would have to feed the corresponding clock into the receiving module as well.

I think initially we had assumed all the escalation sender receive pairs remain in the same domain such that response time is fast. but that obviously is an outdated assumption and cannot be achieved for some sender/receiver combinations in our system...

Reply:
let's place the sync after for now and see how it plays out.
The faster clock tree  doesn't really matter for lc_ctrl since it's not always on.
We can sync up with Nuvoton to see how widespread the effect is.

Reply:
ok, I have added a synchronizer after the esc receiver for the NMI and pulled in the alert handler clock and reset for the escalation receiver primitive.

I also noticed that the alert handler was still sitting in the fast domain, which I think was wrong. the alert handler should reside in the peripheral domain according to our most recent block diagram (https://github.com/lowRISC/opentitan/issues/295). I have hence moved the alert handler into the same clock / reset domains as the timer.

does that look ok to you?